### DISCUSSION TITLE
ts is in hours - how to format for marss?

### DISCUSSION THREAD

**Discussion by emmats** [atsa-es/fish550-2023#34] [Q&A]
The time points in my ts are every 4 hours over 22 days, so the format is date and hour. I think MARSS wants it in years? Should I pretend that every time point is a new year? What is the best way to format my time points for MARSS?

**Comment by eeholmes** [comment]
Hi Emma,

MARSS wants discrete time steps. Unit is unimportant. Note MARSS() will accept a ts object including a multivariate one. So it might be easiest to make your data into a mts object (multivariate ts object).

**Comment by eeholmes** [comment]
Note making the data into a mts object is not important. You just need a matrix shaped like n x T where each time step is 4 hrs. I mentioned the mts option since some data wrangling workflows end up with a ts or mts object. Here is an example of MARSS with a mts.

```
days <- 10
spd <- 6 # samples per day; every 4 hours
n <- 3 # number of time series
dat <- matrix(rnorm(days*spd*n), days*spd, n) # time steps x n matrix
tsdat <- ts(dat, freq=spd)
plot(tsdat)
```
Either of these data formats works:
```
fit <- MARSS(t(dat), method="BFGS") # matrix n x T
fit <- MARSS(tsdat, method="BFGS") # or mts object
```

**Reply by emmats** [reply]
Thank you!
